source/src/cdk/stack-main.ts

```diff
@@ -0,0 +1,214 @@
+      assertions: [{
+        assert: Fn.conditionContains(supportRegions, this.region),
+        assertDescription: 'supported regions are ' + supportRegions.join(', '),
+      }],
+    });
+
+    const prefix = MainStack.SOLUTION_NAME.split(' ').join('-').toLowerCase();
+
+    const {
+      vpc,
+      notebookSg,
+    } = setup_vpc_and_sg(this);
+
+    {
+      const snsKey = new kms.Key(this, 'SNSKey', {
+        enableKeyRotation: true,
+      });
+
+      const iamPolicyStatement = new iam.PolicyStatement({
+        actions: [
+          'kms:Decrypt',
+          'kms:GenerateDataKey',
+        ],
+        principals: [new iam.ServicePrincipal('events.amazonaws.com')],
+        resources: ['*'],
+      });
+
+      snsKey.addToResourcePolicy(iamPolicyStatement);
+
+      snsKey.addToResourcePolicy(new iam.PolicyStatement({
+        actions: [
+          'kms:Decrypt',
+          'kms:GenerateDataKey*',
+          'kms:CreateGrant',
+          'kms:ListGrants',
+          'kms:DescribeKey',
+        ],
+        principals: [new iam.AnyPrincipal()],
+        resources: ['*'],
+        conditions: {
+          StringEquals: {
+            'kms:ViaService': `sns.${this.region}.amazonaws.com`,
+            'kms:CallerAccount': `${this.account}`,
+          },
+        },
+      }));
+      // SNS Topic
+      const topic = new sns.Topic(this, 'SubscriptionsTopic', {
+        masterKey: snsKey,
+      });
+
+      const cfnTopic = topic.node.defaultChild as sns.CfnTopic;
+
+      // eventBridge
+      const eventRule = new events.Rule(this, 'eventRule', {
+        eventPattern: {
+          detail: {
+            status: ['COMPLETED'],
+          },
+          source: ['aws.braket'],
+          detailType: ['Braket Job State Change'],
+        },
+      });
+
+      const subscription = new sns.CfnSubscription(this, 'emailSubcription', {
+        topicArn: cfnTopic.ref,
+        protocol: 'email',
+        endpoint: snsEmail.valueAsString,
+      });
+      subscription.cfnOptions.condition = conditionSnsEmail;
+
+      eventRule.addTarget(new targets.SnsTopic(topic, {
+        message: events.RuleTargetInput.fromText(
+          `Reminder: Job [${EventField.fromPath('$.detail.jobArn')}] is [${EventField.fromPath(
+            '$.detail.status')}], StartedTime:[${EventField.fromPath('$.detail.startedAt')}], FinishedTime:[${EventField.fromPath('$.detail.endedAt')}]`,
+        ),
+      }));
+
+      new CfnOutput(this, 'SNSTopic', {
+        value: topic.topicName,
+        description: `SNS Topic Name(${prefix})`,
+      });
+    }
+
+    const logS3bucket = new s3.Bucket(this, 'accessLogs', {
+      enforceSSL: true,
+      encryption: s3.BucketEncryption.S3_MANAGED,
+    });
+
+    const s3bucket = new s3.Bucket(this, 'amazon-braket', {
+      enforceSSL: true,
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      serverAccessLogsBucket: logS3bucket,
+      serverAccessLogsPrefix: `accessLogs/${logS3bucket.bucketName}/`,
```

Comment:
Why using log bucket name here?

Reply:
It doesn't matter if we use it or not, delete it.